pgo/cmd/auth.go

```diff
@@ -200,96 +198,99 @@ func SetSessionUserCredentials() {
+			return nil, fmt.Errorf("while loading System CA pool - %s", err)
+		} else {
+			caCertPool = pool
+		}
 	}
 
+	// Priority: Flag -> ENV
+	caCertPath := PGO_CA_CERT
 	if caCertPath == "" {
-		fmt.Println("Error: PGO_CA_CERT not specified")
-		os.Exit(2)
-	}
-	caCert, err := ioutil.ReadFile(caCertPath)
-	if err != nil {
-		fmt.Printf("Error: %s", err)
-		fmt.Println("could not read ca certificate")
-		os.Exit(2)
+		caCertPath = os.Getenv("PGO_CA_CERT")
+		if caCertPath == "" {
+			return nil, fmt.Errorf("PGO_CA_CERT not specified")
+		}
 	}
-	caCertPool = x509.NewCertPool()
-	caCertPool.AppendCertsFromPEM(caCert)
 
-	if PGO_CLIENT_CERT != "" {
-		clientCertPath = PGO_CLIENT_CERT
+	// Open trust file and extend trust pool
+	if trustFile, err := os.Open(caCertPath); err != nil {
+		newErr := fmt.Errorf("unable to load TLS trust from %s - [%v]", caCertPath, err)
+		return nil, newErr
 	} else {
-		clientCertPath = os.Getenv("PGO_CLIENT_CERT")
+		err = tlsutil.ExtendTrust(caCertPool, trustFile)
+		if err != nil {
+			newErr := fmt.Errorf("error reading %s - %v", caCertPath, err)
+			return nil, newErr
+		}
+		trustFile.Close()
 	}
 
+	// Priority: Flag -> ENV
+	clientCertPath := PGO_CLIENT_CERT
 	if clientCertPath == "" {
-		fmt.Println("Error: PGO_CLIENT_CERT not specified")
-		os.Exit(2)
-	}
-
-	_, err = ioutil.ReadFile(clientCertPath)
-	if err != nil {
-		log.Debugf("%s not found", clientCertPath)
-		os.Exit(2)
-	}
-
-	if PGO_CLIENT_KEY != "" {
-		clientKeyPath = PGO_CLIENT_KEY
-	} else {
-		clientKeyPath = os.Getenv("PGO_CLIENT_KEY")
+		clientCertPath = os.Getenv("PGO_CLIENT_CERT")
+		if clientCertPath == "" {
+			return nil, fmt.Errorf("PGO_CLIENT_CERT not specified")
+		}
 	}
 
+	// Priority: Flag -> ENV
+	clientKeyPath := PGO_CLIENT_KEY
 	if clientKeyPath == "" {
-		fmt.Println("Error: PGO_CLIENT_KEY not specified")
-		os.Exit(2)
+		clientKeyPath = os.Getenv("PGO_CLIENT_KEY")
+		if clientKeyPath == "" {
+			return nil, fmt.Errorf("PGO_CLIENT_KEY not specified")
+		}
 	}
 
-	_, err = ioutil.ReadFile(clientKeyPath)
+	certPair, err := tls.LoadX509KeyPair(clientCertPath, clientKeyPath)
 	if err != nil {
-		log.Debugf("%s not found", clientKeyPath)
-		os.Exit(2)
-	}
-	cert, err = tls.LoadX509KeyPair(clientCertPath, clientKeyPath)
-	if err != nil {
-		fmt.Printf("Error loading client certificate/key: %s\n", err)
-		os.Exit(2)
+		return nil, fmt.Errorf("client certificate/key loading: %s", err)
 	}
 
 	// create a Transport object for use by the HTTP client
 	return &http.Transport{
 		TLSClientConfig: &tls.Config{
 			RootCAs:            caCertPool,
 			InsecureSkipVerify: true,
-			Certificates:       []tls.Certificate{cert},
+			Certificates:       []tls.Certificate{certPair},
 			MinVersion:         tls.VersionTLS11,
 		},
+	}, nil
+}
+
+// NewAPIClient returns an http client configured with a tls.Config
+// based on environmental settings and a default timeout
+func NewAPIClient() *http.Client {
+	defaultTimeout := 60 * time.Second
```

Comment:
Is it really worth declaring `defaultTimeout` here if it isn't being used in multiple places?

Reply:
I debated that a lot, honestly. Ultimately I decided to leave it as a defined variable so that the literal definition following looks less "magical", more self-documenting, and set up to be refactored out if knowledge of the duration is needed elsewhere (or if it needs to be parameterized). Said differently, I settled on it to communicate the 60 was a 'default' choice and not some hard-coded value with undocumented dependency.

At this point, I'll leave it as-is unless someone else has a stronger opinion than you or I.